api/representation_test.go

```diff
@@ -47,6 +47,289 @@ func TestTaskRequest_String(t *testing.T) {
 	require.Equal(t, expected, actual)
 }
 
+func TestRequest_TaskRequestFromTaskConfig(t *testing.T) {
```

Comment:
Yeah, I could rename this to `Test_oapigenTaskFromConfigTask` instead, and then write a simple test for `TestTaskResponse_taskResponseFromTaskConfig` which also calls `oapigenTaskFromConfigTask`.

As for testing the variables, I opted to test it in the e2e tests because of the file requirements. So unless we refactor to maybe use an io.Reader instead, it's not really unit testable. I can look into doing that?

Reply:
Hmmm. There isn't much complexity to the Variables implementation but in general it could be nice to have coverage and exercise / check the specific functionality. I find e2e test great for verifying all the functionality works together to produce a reasonable outcome. I think it could be worth time-boxing a small amount to see if its reasonably easy to make it unit-testable. If not, I'm ok with not having a `TestTaskResponse_TaskResponse_FromTaskConfig`.

That's my thought process. Open to your thoughts

Reply:
Ok, sounds good, I'll give it a shot

Reply:
I like the abstracting out io.Reader you went with!